vllm/sequence.py

```diff
@@ -203,35 +215,64 @@ def __init__(
         arrival_time: float,
     ) -> None:
         self.request_id = request_id
-        self.seqs = seqs
+        self.seqs_dict = {seq.seq_id: seq for seq in seqs}
         self.sampling_params = sampling_params
         self.arrival_time = arrival_time
 
+    def get_max_num_running_seqs(self) -> int:
+        """The maximum number of sequences running in parallel in the remaining
+        lifetime of the request."""
+        if self.sampling_params.use_beam_search:
+            # For beam search, there will always be `best_of` sequences running
```

Comment:
To my understanding of the discussion we had offline, this comment is not correct, right?

Reply:
Thanks for catching this! Fixed with a new version of the comment.